Pop'n Music 27 Unilab support

jorpz opened the pull request:
verifylint ✔️
verifytyping ✔️
formatfiles 🍰
trafficgen ✔️
verifytraffic ✔️

No major changes from 26 aside from phase stuff and some new profile stuff.

Known issues:
I don't know how to trigger KAC Lab. This seems to be something that should be able to be accessed on appropriate versions of the dll but I can't seem to figure it out. Rare softlock on pop'n quest Lively II event if you mess with the phase flags and put the game in an invalid state. In theory (and according to bemaniwiki) the entire event should be clearable on earlier Unilab builds.

Not an issue/will not fix:
狼弦暴威 does not appear in Awakening Elem when the event flag is set. The solution to this (for some reason) is to clear the other 10 events. This is not a bemaniutils issue.

jorpz committed:
Pop'n Music 27 Unilab support

Known issues:
I don't know how to trigger KAC Lab. This seems to be something that should be able to be accessed on appropriate versions of the dll but I can't seem to figure it out.
Rare softlock on pop'n quest Lively II event if you mess with the phase flags and put the game in an invalid state. In theory (and according to bemaniwiki) the entire event should be clearable on earlier Unilab builds.

Not an issue/will not fix:
狼弦暴威 does not appear in Awakening Elem when the event flag is set. The solution to this (for some reason) is to clear the other 10 events. This is not a bemaniutils issue.

DragonMinded commented on bemani/backend/popn/unilab.py:
```diff
@@ -0,0 +1,640 @@
+
+        # Set up Kakusei no Elem achievements
+        game_config = self.get_game_config()
+        if game_config.get_bool("force_unlock_deco"):
```
We are setting maximum energy and battery here, but in the `unformat_profile` function we don't ignore updating this value if `force_unlock_deco` is true, which will poison user profiles with fully unlocked events.

DragonMinded commented on bemani/backend/popn/unilab.py:
```diff
@@ -0,0 +1,640 @@
+                )
+
+            # Extract Kakusei no Elem achievements
+            battery_data = event_p27.child("battery")
```
Should skip this whole section if `force_unlock_deco` is true.

DragonMinded reviewed (commented):
Aside from the comments this looks good to merge.

jorpz committed:
Avoid progression lockout if deco parts are unintentionally force unlocked and updated in the achivements table for pop'n Unilab and pop'n Kaimei Riddles

jorpz committed:
Revert "Avoid progression lockout if deco parts are unintentionally force unlocked and updated in the achivements table for pop'n Unilab and pop'n Kaimei Riddles"

This reverts commit 56068edf37f8ee87d3b4355f4d4556fff6a4b9b4.

jorpz committed:
Protect profile from unilab deco force unlock

Avoid progression lockout if deco parts are unintentionally force unlocked and updated in the achievements table for pop'n Unilab
